frontend/src/metabase-lib/queries/utils/dataset.ts

```diff
@@ -1,93 +1,46 @@
-import _ from "underscore";
-
-import * as Lib from "metabase-lib";
-import Dimension from "metabase-lib/Dimension";
+import { getColumnKey } from "metabase-lib/queries/utils/get-column-key";
 import { normalize } from "metabase-lib/queries/utils/normalize";
 import type {
   DatasetColumn,
   DatasetData,
-  FieldReference,
   TableColumnOrderSetting,
 } from "metabase-types/api";
 
 export const datasetContainsNoResults = (data: DatasetData) =>
   data.rows == null || data.rows.length === 0;
 
-/**
- * Returns a MBQL field reference (FieldReference) for a given result dataset column
- */
-export function fieldRefForColumn(
-  column: DatasetColumn,
-): FieldReference | null | undefined {
-  // NOTE: matching existing behavior of returning the unwrapped base dimension until we understand the implications of changing this
-  return (
-    column.field_ref &&
-    Dimension.parseMBQL(column.field_ref)?.baseDimension().mbql()
-  );
+export function getColumnSettingKey({
```

Comment:
I assume the backend implementation for the table static viz should match this to have entirely synced behavior. If I'm right, can you please create an issue and provide an example where it did not work so that BE engineers can work on it? cc @markbastian @adam-james-v